src/services/ivaDefectsService.ts

```diff
@@ -3,58 +3,58 @@ import {
   InspectionType,
   VehicleType,
 } from "@dvsa/cvs-type-definitions/types/iva/defects/get";
-import {EUVehicleCategory} from "@dvsa/cvs-type-definitions/types/v3/tech-record/enums/euVehicleCategory.enum";
+import { EUVehicleCategory } from "@dvsa/cvs-type-definitions/types/v3/tech-record/enums/euVehicleCategory.enum";
+import { IvaDatabaseService } from "./ivaDatabaseService";
+import { HTTPError } from "../models/HTTPError";
+import { convertFlatDataToProperJSON } from "../utils/formatDefects";
 
 export class IvaDefectsService {
+  public readonly ivaDatabaseService: IvaDatabaseService;
+
+  constructor(ivaDatabaseService: IvaDatabaseService) {
+    this.ivaDatabaseService = ivaDatabaseService;
+  }
+
   public getIvaDefects(
     vehicleType: VehicleType | null,
     euVehicleCategory: EUVehicleCategory | null,
     inspectionType: InspectionType | null,
   ): Promise<DefectGETIVA[]> {
-    return new Promise((resolve) => {
-      const mockIvaDefects: DefectGETIVA[] = [
-        {
-          sectionNumber: "01",
-          sectionDescription: "Noise",
-          vehicleTypes: ["hgv"],
-          euVehicleCategories: [EUVehicleCategory.M1],
-          requiredStandards: [
-            {
-              rsNumber: 1,
-              requiredStandard: "A mock standard",
-              refCalculation: "1.1",
-              additionalInfo: true,
-              inspectionTypes: ["basic"],
-            },
-          ],
-        },
-      ];
-
-      resolve(mockIvaDefects);
-    });
+    return this.ivaDatabaseService
```

Comment:
Is it worth this method being written using async await too, like the method below.

Not sure what everyone else thinks, but I find async await more readable.

Reply:
I always prefer async await to promises - but that might just be the C# Fanboy in me. Certainly think consistency within the same service is worthwhile, whichever we go with.

Reply:
I'm the same, but that is also likely the Java Fanboy in me. Yes, either way they should both be the same.

Reply:
agree from me too